sql/catalog/lease: TestKVWriterMatchesIEWriter failed

sql/catalog/lease.TestKVWriterMatchesIEWriter [failed](https://tanzanite.cluster.engflow.com/invocation/141d254c-e218-4b41-8fd8-0908c77c2bed) on master @ [a1f68527966c9b5c206e26b785cc03be24d1a722](https://github.com/cockroachdb/cockroach/commits/a1f68527966c9b5c206e26b785cc03be24d1a722):


```
        	            	+  Value: (roachpb.Value) {
        	            	+   RawBytes: ([]uint8) (len=5) {
        	            	+    00000000  00 00 00 00 0a                                    |.....|
        	            	+   },
        	            	+   Timestamp: (hlc.Timestamp) {
        	            	+    WallTime: (int64) 0,
        	            	+    Logical: (int32) 0,
        	            	+    Synthetic: (bool) false
        	            	+   }
        	            	+  }
        	            	+ },
        	            	+ (roachpb.KeyValue) {
        	            	+  Key: (roachpb.Key) (len=21) {
        	            	+   00000000  8a 12 80 00 01 91 91 14  fa 01 cc 5f ae c5 f9 19  |..........._....|
        	            	+   00000010  f2 b3 40 8e 88                                    |..@..|
        	            	+  },
        	            	+  Value: (roachpb.Value) {
        	            	+   RawBytes: ([]uint8) (len=5) {
        	            	+    00000000  00 00 00 00 0a                                    |.....|
        	            	+   },
        	            	+   Timestamp: (hlc.Timestamp) {
        	            	+    WallTime: (int64) 0,
        	            	+    Logical: (int32) 0,
        	            	+    Synthetic: (bool) false
        	            	+   }
        	            	+  }
        	            	+ },
        	            	+ (roachpb.KeyValue) {
        	            	+  Key: (roachpb.Key) (len=21) {
        	            	+   00000000  8a 12 80 00 01 91 91 14  fa 02 10 4c 71 41 f9 13  |...........LqA..|
        	            	+   00000010  3c f8 f8 90 88                                    |<....|
        	            	+  },
        	            	+  Value: (roachpb.Value) {
        	            	+   RawBytes: ([]uint8) (len=5) {
        	            	+    00000000  00 00 00 00 0a                                    |.....|
        	            	+   },
        	            	+   Timestamp: (hlc.Timestamp) {
        	            	+    WallTime: (int64) 0,
        	            	+    Logical: (int32) 0,
        	            	+    Synthetic: (bool) false
        	            	+   }
        	            	+  }
        	            	+ },
        	            	+ (roachpb.KeyValue) {
        	            	+  Key: (roachpb.Key) (len=21) {
        	            	+   00000000  8a 12 80 00 01 91 91 14  fa 02 17 71 0f 17 f9 33  |...........q...3|
        	            	+   00000010  2f be 98 8f 88                                    |/....|
        	            	   },
        	Test:       	TestKVWriterMatchesIEWriter/Session_based_leasing_mode:_3
    --- FAIL: TestKVWriterMatchesIEWriter/Session_based_leasing_mode:_3 (4.12s)
```
<p>Parameters: <code>attempt=1</code>
, <code>run=17</code>
, <code>shard=1</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKVWriterMatchesIEWriter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-33802

sql/catalog/lease.TestKVWriterMatchesIEWriter [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12802310?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12802310?buildTab=artifacts#/) on master @ [59fb4ec268b456b7c5ad694c43202731f3299efe](https://github.com/cockroachdb/cockroach/commits/59fb4ec268b456b7c5ad694c43202731f3299efe):


```
        	            	+  },
        	            	+  Value: (roachpb.Value) {
        	            	+   RawBytes: ([]uint8) (len=5) {
        	            	+    00000000  00 00 00 00 0a                                    |.....|
        	            	+   },
        	            	+   Timestamp: (hlc.Timestamp) {
        	            	+    WallTime: (int64) 0,
        	            	+    Logical: (int32) 0,
        	            	+    Synthetic: (bool) false
        	            	+   }
        	            	+  }
        	            	+ },
        	            	+ (roachpb.KeyValue) {
        	            	+  Key: (roachpb.Key) (len=21) {
        	            	+   00000000  8a 12 80 00 01 91 91 14  fa 01 ba 9e 3d a7 f9 1f  |............=...|
        	            	+   00000010  e4 7d 60 8a 88                                    |.}``..|
        	            	+  },
        	            	+  Value: (roachpb.Value) {
        	            	+   RawBytes: ([]uint8) {
        	            	+   },
        	            	+   Timestamp: (hlc.Timestamp) {
        	            	+    WallTime: (int64) 0,
        	            	+    Logical: (int32) 0,
        	            	+    Synthetic: (bool) false
        	            	+   }
        	            	+  }
        	            	+ },
        	            	+ (roachpb.KeyValue) {
        	            	+  Key: (roachpb.Key) (len=21) {
        	            	+   00000000  8a 12 80 00 01 91 91 14  fa 01 ba 9e 3d a7 f9 1f  |............=...|
        	            	+   00000010  e4 7d 60 8a 88                                    |.}``..|
        	            	+  },
        	            	+  Value: (roachpb.Value) {
        	            	+   RawBytes: ([]uint8) (len=5) {
        	            	+    00000000  00 00 00 00 0a                                    |.....|
        	            	+   },
        	            	+   Timestamp: (hlc.Timestamp) {
        	            	+    WallTime: (int64) 0,
        	            	+    Logical: (int32) 0,
        	            	+    Synthetic: (bool) false
        	            	+   }
        	            	+  }
        	            	+ },
        	            	+ (roachpb.KeyValue) {
        	            	+  Key: (roachpb.Key) (len=21) {
        	            	+   00000000  8a 12 80 00 01 91 91 14  fa 02 21 4f 6c 8b f9 13  |..........!Ol...|
        	            	+   00000010  3e 1a 08 8d 88                                    |>....|
        	            	   },
        	Test:       	TestKVWriterMatchesIEWriter/Session_based_leasing_mode:_3
    --- FAIL: TestKVWriterMatchesIEWriter/Session_based_leasing_mode:_3 (19.73s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKVWriterMatchesIEWriter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/catalog/lease.TestKVWriterMatchesIEWriter [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12805343?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12805343?buildTab=artifacts#/) on master @ [247823ac3cde898b276677b59cd1617ce94dc6e7](https://github.com/cockroachdb/cockroach/commits/247823ac3cde898b276677b59cd1617ce94dc6e7):


```
        	            	+  Value: (roachpb.Value) {
        	            	+   RawBytes: ([]uint8) (len=5) {
        	            	+    00000000  00 00 00 00 0a                                    |.....|
        	            	+   },
        	            	+   Timestamp: (hlc.Timestamp) {
        	            	+    WallTime: (int64) 0,
        	            	+    Logical: (int32) 0,
        	            	+    Synthetic: (bool) false
        	            	+   }
        	            	+  }
        	            	+ },
        	            	+ (roachpb.KeyValue) {
        	            	+  Key: (roachpb.Key) (len=21) {
        	            	+   00000000  8a 12 80 00 01 91 91 14  fa 02 03 79 84 3e f9 20  |...........y.>. |
        	            	+   00000010  57 63 70 8c 88                                    |Wcp..|
        	            	+  },
        	            	+  Value: (roachpb.Value) {
        	            	+   RawBytes: ([]uint8) (len=5) {
        	            	+    00000000  00 00 00 00 0a                                    |.....|
        	            	+   },
        	            	+   Timestamp: (hlc.Timestamp) {
        	            	+    WallTime: (int64) 0,
        	            	+    Logical: (int32) 0,
        	            	+    Synthetic: (bool) false
        	            	+   }
        	            	+  }
        	            	+ },
        	            	+ (roachpb.KeyValue) {
        	            	+  Key: (roachpb.Key) (len=21) {
        	            	+   00000000  8a 12 80 00 01 91 91 14  fa 02 08 ab 1a c0 f9 11  |................|
        	            	+   00000010  e4 1f b8 8f 88                                    |.....|
        	            	+  },
        	            	+  Value: (roachpb.Value) {
        	            	+   RawBytes: ([]uint8) (len=5) {
        	            	+    00000000  00 00 00 00 0a                                    |.....|
        	            	+   },
        	            	+   Timestamp: (hlc.Timestamp) {
        	            	+    WallTime: (int64) 0,
        	            	+    Logical: (int32) 0,
        	            	+    Synthetic: (bool) false
        	            	+   }
        	            	+  }
        	            	+ },
        	            	+ (roachpb.KeyValue) {
        	            	+  Key: (roachpb.Key) (len=21) {
        	            	+   00000000  8a 12 80 00 01 91 91 14  fa 02 1e 1e bb d7 f9 31  |...............1|
        	            	+   00000010  50 96 88 8a 88                                    |P....|
        	            	   },
        	Test:       	TestKVWriterMatchesIEWriter/Session_based_leasing_mode:_3
    --- FAIL: TestKVWriterMatchesIEWriter/Session_based_leasing_mode:_3 (8.89s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKVWriterMatchesIEWriter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/catalog/lease.TestKVWriterMatchesIEWriter [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12806780?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12806780?buildTab=artifacts#/) on master @ [23a7242ece434c4dcec48953010621fa6bdb7824](https://github.com/cockroachdb/cockroach/commits/23a7242ece434c4dcec48953010621fa6bdb7824):


```
        	            	+  Value: (roachpb.Value) {
        	            	+   RawBytes: ([]uint8) (len=5) {
        	            	+    00000000  00 00 00 00 0a                                    |.....|
        	            	+   },
        	            	+   Timestamp: (hlc.Timestamp) {
        	            	+    WallTime: (int64) 0,
        	            	+    Logical: (int32) 0,
        	            	+    Synthetic: (bool) false
        	            	+   }
        	            	+  }
        	            	+ },
        	            	+ (roachpb.KeyValue) {
        	            	+  Key: (roachpb.Key) (len=21) {
        	            	+   00000000  8a 12 80 00 01 91 91 14  fa 01 9d 4e f3 1b f9 05  |...........N....|
        	            	+   00000010  bd 29 78 8a 88                                    |.)x..|
        	            	+  },
        	            	+  Value: (roachpb.Value) {
        	            	+   RawBytes: ([]uint8) (len=5) {
        	            	+    00000000  00 00 00 00 0a                                    |.....|
        	            	+   },
        	            	+   Timestamp: (hlc.Timestamp) {
        	            	+    WallTime: (int64) 0,
        	            	+    Logical: (int32) 0,
        	            	+    Synthetic: (bool) false
        	            	+   }
        	            	+  }
        	            	+ },
        	            	+ (roachpb.KeyValue) {
        	            	+  Key: (roachpb.Key) (len=21) {
        	            	+   00000000  8a 12 80 00 01 91 91 14  fa 01 d6 0a d3 d0 f9 04  |................|
        	            	+   00000010  04 2e d8 91 88                                    |.....|
        	            	+  },
        	            	+  Value: (roachpb.Value) {
        	            	+   RawBytes: ([]uint8) (len=5) {
        	            	+    00000000  00 00 00 00 0a                                    |.....|
        	            	+   },
        	            	+   Timestamp: (hlc.Timestamp) {
        	            	+    WallTime: (int64) 0,
        	            	+    Logical: (int32) 0,
        	            	+    Synthetic: (bool) false
        	            	+   }
        	            	+  }
        	            	+ },
        	            	+ (roachpb.KeyValue) {
        	            	+  Key: (roachpb.Key) (len=21) {
        	            	+   00000000  8a 12 80 00 01 91 91 14  fa 02 0a 68 34 4d f9 24  |...........h4M.$|
        	            	+   00000010  34 91 e8 8e 88                                    |4....|
        	            	   },
        	Test:       	TestKVWriterMatchesIEWriter/Session_based_leasing_mode:_3
    --- FAIL: TestKVWriterMatchesIEWriter/Session_based_leasing_mode:_3 (7.12s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKVWriterMatchesIEWriter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/catalog/lease.TestKVWriterMatchesIEWriter [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12806839?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12806839?buildTab=artifacts#/) on master @ [23a7242ece434c4dcec48953010621fa6bdb7824](https://github.com/cockroachdb/cockroach/commits/23a7242ece434c4dcec48953010621fa6bdb7824):


```
        	            	+  },
        	            	+  Value: (roachpb.Value) {
        	            	+   RawBytes: ([]uint8) (len=5) {
        	            	+    00000000  00 00 00 00 0a                                    |.....|
        	            	+   },
        	            	+   Timestamp: (hlc.Timestamp) {
        	            	+    WallTime: (int64) 0,
        	            	+    Logical: (int32) 0,
        	            	+    Synthetic: (bool) false
        	            	+   }
        	            	+  }
        	            	+ },
        	            	+ (roachpb.KeyValue) {
        	            	+  Key: (roachpb.Key) (len=21) {
        	            	+   00000000  8a 12 80 00 01 91 91 14  fa 01 e4 54 93 0c f9 0d  |...........T....|
        	            	+   00000010  14 15 60 8f 88                                    |..``..|
        	            	+  },
        	            	+  Value: (roachpb.Value) {
        	            	+   RawBytes: ([]uint8) {
        	            	+   },
        	            	+   Timestamp: (hlc.Timestamp) {
        	            	+    WallTime: (int64) 0,
        	            	+    Logical: (int32) 0,
        	            	+    Synthetic: (bool) false
        	            	+   }
        	            	+  }
        	            	+ },
        	            	+ (roachpb.KeyValue) {
        	            	+  Key: (roachpb.Key) (len=21) {
        	            	+   00000000  8a 12 80 00 01 91 91 14  fa 01 e4 54 93 0c f9 0d  |...........T....|
        	            	+   00000010  14 15 60 8f 88                                    |..``..|
        	            	+  },
        	            	+  Value: (roachpb.Value) {
        	            	+   RawBytes: ([]uint8) (len=5) {
        	            	+    00000000  00 00 00 00 0a                                    |.....|
        	            	+   },
        	            	+   Timestamp: (hlc.Timestamp) {
        	            	+    WallTime: (int64) 0,
        	            	+    Logical: (int32) 0,
        	            	+    Synthetic: (bool) false
        	            	+   }
        	            	+  }
        	            	+ },
        	            	+ (roachpb.KeyValue) {
        	            	+  Key: (roachpb.Key) (len=21) {
        	            	+   00000000  8a 12 80 00 01 91 91 14  fa 01 ea db 0d 29 f9 34  |.............).4|
        	            	+   00000010  3d b0 10 89 88                                    |=....|
        	            	   },
        	Test:       	TestKVWriterMatchesIEWriter/Session_based_leasing_mode:_3
    --- FAIL: TestKVWriterMatchesIEWriter/Session_based_leasing_mode:_3 (6.85s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKVWriterMatchesIEWriter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/catalog/lease/kv_writer_test.go
+++ b/pkg/sql/catalog/lease/kv_writer_test.go
@@ -64,10 +64,10 @@ func TestKVWriterMatchesIEWriter(t *testing.T) {
 			ctx := context.Background()
 			serverArgs := base.TestServerArgs{}
 			serverArgs.Settings = cluster.MakeClusterSettings()
-			LeaseEnableSessionBasedLeasing.Override(ctx, &serverArgs.Settings.SV, int64(mode))
 			srv, sqlDB, kvDB := serverutils.StartServer(t, serverArgs)
 			defer srv.Stopper().Stop(ctx)
 			s := srv.ApplicationLayer()
+			LeaseEnableSessionBasedLeasing.Override(ctx, &s.ClusterSettings().SV, int64(mode))
 
 			// Otherwise, we wouldn't get complete SSTs in our export under stress.
 			sqlutils.MakeSQLRunner(srv.SystemLayer().SQLConn(t)).Exec(
@@ -102,6 +102,7 @@ func TestKVWriterMatchesIEWriter(t *testing.T) {
 				a: newInternalExecutorWriter(ie, "defaultdb.public.lease1", mode),
 				b: newKVWriter(codec, kvDB, lease2ID, settingsWatcher, modeReader),
 			}
+			require.Equal(t, modeReader.getSessionBasedLeasingMode(), mode)
 			start := kvDB.Clock().Now()
 			groups := generateWriteOps(2<<10, 1<<10)
 			for {
@@ -230,7 +231,7 @@ func generateWriteOps(n, numGroups int) func() (_ []writeOp, wantMore bool) {
 			version:      descpb.DescriptorVersion(rand.Intn(vals)),
 			instanceID:   base.SQLInstanceID(rand.Intn(vals)),
 			expiration:   *ts,
-			sessionID:    []byte(ts.String()),
+			sessionID:    []byte(ts.String() + "_session"),
 			regionPrefix: enum.One,
 		}
 		return lf

